analyze_visual/wrapper.py

```diff
@@ -82,10 +92,24 @@ def main(argv):
             features_stats = process_video(v, 2, True, True, True)
             features = features_stats[0]
             features = features.reshape(1, -1)
-            # Predict the classes of shots
-            r = video_class_predict(features, algorithm)
-            results[v]= r
-    [print(f'Video {key} belongs to {value}') for key, value in results.items()]    
+            probas, classes = video_class_predict(features, algorithm)
+            #Save the resuls in a numpy array
+            final_proba = np.append(final_proba,[probas],axis=0)
+            #Save results of every video in a text file
+            with open(str(videos_path)+".txt", "a") as text_file:
```

Comment:
videos_path is a directory. This could cause issues if ends in /.
Anyway, please add the output csv file as an argument (-o)
Also, output should be of csv format with columns:
filename, <class_1_name> percentage, <class_2_name> percentage> 
..., ..., ..., ...

And you can only print the averages after the processing finishes